tower-http/src/decompression/request/layer.rs

```diff
@@ -0,0 +1,93 @@
+use super::service::RequestDecompression;
+use crate::compression_utils::AcceptEncoding;
+use tower_layer::Layer;
+
+/// Decompresses request bodies and calls its underlying service.
+///
+/// Transparently decompresses request bodies based on the `Content-Encoding` header.
+/// When the encoding in the `Content-Encoding` header is not accepted an `Unsupported Media Type`
+/// status code will be returned with the accepted encodings in the `Accept-Encoding` header.
+///
+/// Enabling pass-through of unaccepted encodings will not return an `Unsupported Media Type`. But
+/// will call the underlying service with the unmodified request if the encoding is not supported.
+/// This is disabled by default.
+///
+/// See the [module docs](crate::decompression) for more details.
+#[derive(Debug, Default, Clone)]
+pub struct RequestDecompressionLayer {
+    accept: AcceptEncoding,
+    pass_through_unaccepted: bool,
+}
+
+impl<S> Layer<S> for RequestDecompressionLayer {
+    type Service = RequestDecompression<S>;
+
+    fn layer(&self, service: S) -> Self::Service {
+        RequestDecompression {
+            inner: service,
+            accept: self.accept,
+            pass_through_unaccepted: self.pass_through_unaccepted,
+        }
+    }
+}
+
+impl RequestDecompressionLayer {
+    /// Creates a new `RequestDecompressionLayer`.
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    /// Sets whether to support
+    /// gzip encoding.
```

Comment:
Some odd line breaks going on here and in the other methods as well.

Reply:
Not sure what I did there.